### DISCUSSION TITLE
Restart container on loss of network connection

### DISCUSSION THREAD

**Discussion by gabrielwhite** [bubuntux/nordvpn#195] [Show and tell]
I've had repeated trouble of containers that use NordVPN losing their network connection when the NordVPN connection itself restarts (which seems to happen 1-2 times a day).

Here's a script I made to do a healthcheck every 3 minutes, and restart the container if needed:

```
# First check that the containers are running

if ! docker ps | grep nordvpn &>/dev/null; then
  echo "NordVPN is not running"
  exit 1
elif ! docker ps | grep othercontainer &>/dev/null; then
  echo "Othercontainer is not running"
  exit 1
fi

# Test if othercontainer has an internet connection. If not, restart the container

if ! docker exec -i othercontainer curl -f -s -I "https://www.google.com/" &>/dev/null && true; then
  docker restart othercontainer
fi
```

**Comment by bubuntux** [comment]
Give it a try to the latest version, it may reconnect but the container should not restart keeping the connection to the others containers alive.

**Reply by gabrielwhite** [reply]
Seems to have fixed the issue thanks! I've seen two reconnects in NordVPN and my checking script hasn't done any restarts.
